ReactSkia/platform/linux/RuntimeEventBeat.cpp

```diff
@@ -5,23 +5,67 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+#include <thread>
+
 #include "RuntimeEventBeat.h"
 #include "ReactSkia/utils/RnsLog.h"
 
+#define BEAT_INTERVAL  1 /*Temp:1 sec*/
+
 namespace facebook {
 namespace react {
 
-RuntimeEventBeat::RuntimeEventBeat(EventBeat::SharedOwnerBox const &ownerBox, RuntimeExecutor runtimeExecutor)
-    : EventBeat(ownerBox), runtimeExecutor_(std::move(runtimeExecutor)) {
-    RNS_LOG_NOT_IMPL;
+namespace {
+
+/*
+EventBeatThread : to push the dispatched asynchronous event to the JS world
+TODO: BEAT_INTERVAL to be refined to the need.
+*/
+void BeatThread(RuntimeEventBeat *runtimeEventBeat,RunLoopObserver::Activity activities) {
+  while(RuntimeEventBeat::keepBeating) {
+    sleep(BEAT_INTERVAL);
+    runtimeEventBeat->triggerBeat(activities);
+  }
+}
+
+}//namespace
+
+bool RuntimeEventBeat::keepBeating{true};
+/*Activity is dummy param to fit in React common implementation, It is not needed as
+  RNS implementation is not based on runloop*/
+RunLoopObserver::Activity activities{RunLoopObserver::Activity::BeforeWaiting};
+RuntimeEventBeat::RuntimeEventBeat(RunLoopObserver::WeakOwner const &owner)
+  : RunLoopObserver(activities, owner) {
+  /*Spawning  Beat Thread*/
+  std::thread thread(BeatThread,this,activities);
+  thread.detach();
 }
 
 RuntimeEventBeat::~RuntimeEventBeat() {
-    RNS_LOG_NOT_IMPL;
+  RuntimeEventBeat::keepBeating=false; /*Signal BeatThread to stop Beating*/
+}
+
+void RuntimeEventBeat::triggerBeat(RunLoopObserver::Activity activities){
+  this->activityDidChange(activities);
+}
+void RuntimeEventBeat::startObserving() const noexcept
+{
+  RNS_LOG_NOT_IMPL;
+  /*Start Beating once the Beat thread created*/
+}
+
+void RuntimeEventBeat::stopObserving() const noexcept
+{
+  RNS_LOG_NOT_IMPL;
+  /*TODO:Need to check the usecase for stop & start beating
+         Based on the Beat thread to be redesigned*/
 }
 
-void RuntimeEventBeat::induce() const {
-    RNS_LOG_NOT_IMPL;
+bool RuntimeEventBeat::isOnRunLoopThread() const noexcept
+{
+  /*As Beat thread is not based on runLoop, having dummy implementation to fit in to
+    React common runlooponsrever design need*/
+  return true;
```

Comment:
Here we have to check the thread id ? 
thread id of the caller thread and event beat thread ?